src/sourceFetch.js

```diff
@@ -0,0 +1,26 @@
+import ErrorPopup from "./errorHandler.js";
+import IntilizerClass from "./SourceFetchService.js";
+export default class SourceFetcher {
+  static async fetch(url) {
+    try {
+      const response = await fetch(url);
+      if (response.status === 200) {
+        const data = await response.json();
+        SourceFetcher.NewssourceSelectComponent(data);
```

Comment:
that is also better to do not in the scope of fetcher. better to return a promise and then do all that stuffs in component that tries to fetch smth